{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: BSgenome\n",
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colMeans,\n",
      "    colnames, colSums, dirname, do.call, duplicated, eval, evalq,\n",
      "    Filter, Find, get, grep, grepl, intersect, is.unsorted, lapply,\n",
      "    lengths, Map, mapply, match, mget, order, paste, pmax, pmax.int,\n",
      "    pmin, pmin.int, Position, rank, rbind, Reduce, rowMeans, rownames,\n",
      "    rowSums, sapply, setdiff, sort, table, tapply, union, unique,\n",
      "    unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "Loading required package: stats4\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "Loading required package: GenomeInfoDb\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: Biostrings\n",
      "Loading required package: XVector\n",
      "\n",
      "Attaching package: ‘Biostrings’\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    strsplit\n",
      "\n",
      "Loading required package: rtracklayer\n"
     ]
    }
   ],
   "source": [
    "library(BSgenome.Hsapiens.UCSC.hg38)\n",
    "library(data.table)\n",
    "for (i in names(Hsapiens)[1:24]){ # Load L1 target site annotation files\n",
    "        load(paste0('../sim-develop/data/root_maps/',i,'.rda'))\n",
    "}\n",
    "library(EnsDb.Hsapiens.v86)\n",
    "\n",
    "gann <- genes(EnsDb.Hsapiens.v86)\n",
    "gann <- data.table(chrom=as.vector(seqnames(gann)),start=start(gann),end=end(gann),geneSym=gann$gene_name)\n",
    "head(gann)\n",
    "\n",
    "system(paste(\"rm ./count_all_sites_in_genes.log\"))\n",
    "outPath <- '~/jackgl/gene_all_counts/out'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "for (ii in names(Hsapiens)[1:24]){ # Loop over chromosomes\n",
    "\n",
    "    ptm <- proc.time()\n",
    "    line <- paste0('######## Chromosome ',ii,' #########\\n')\n",
    "    write(line,file=paste0(outPath,'.log'),append=TRUE)\n",
    "    \n",
    "    map<-get(paste0(ii,'Map')) # Get the L1 target site annotation for the current chromosome\n",
    "    ict<-    map[[2]]\n",
    "    icl<-    map[[3]]\n",
    "    iot<-    map[[4]]\n",
    "    iol<-    map[[5]]\n",
    "    insites<-map[[1]]\n",
    "\n",
    "    chrno <- strsplit(ii,\"chr\")[[1]][2] # Get the current chromosome symbol\n",
    "    tmpann_1 <- gann[gann$chrom==chrno,] # Extract gann entries for current chromosome\n",
    "    currGenes <- unique(tmpann_1$geneSym) # Get list of genes in current chromosome\n",
    "    numCurrGenes <- length(currGenes) # Get length of the list\n",
    "    counts <-  array(NA,dim=c(numCurrGenes,4)) # Allocate matrix for counts\n",
    "    gnames <- array(NA,dim=c(numCurrGenes,1))\n",
    "    l<-1\n",
    "\n",
    "    for (jj in currGenes) { # Loop over genes of current chromosome\n",
    "\n",
    "        line <- paste0('gene ',l,'/',numCurrGenes,' for ',ii,':\\t',jj)\n",
    "        write(line,file=paste0(outPath,'.log'),append=TRUE)\n",
    "        \n",
    "        # count sites in exons of current gene\n",
    "        tmpann_2 <- tmpann_1[tmpann_1$geneSym==jj,] # Extract gann entries for current gene\n",
    "\n",
    "        sen        <-inrange(insites[ict[!is.na(ict[,1]),1],1],tmpann_2$start,tmpann_2$end) # Check if any Closed-Tight category sites are within the start-end range of tmpann_2\n",
    "        antisen    <-inrange(insites[ict[!is.na(ict[,2]),2],2],tmpann_2$start,tmpann_2$end)\n",
    "        counts[l,1]<-length(which(sen)) + length(which(antisen))                            # Fill an element of the counts table\n",
    "\n",
    "        sen        <-inrange(insites[icl[!is.na(icl[,1]),1],1],tmpann_2$start,tmpann_2$end)\n",
    "        antisen    <-inrange(insites[icl[!is.na(icl[,2]),2],2],tmpann_2$start,tmpann_2$end)\n",
    "        counts[l,2]<-length(which(sen)) + length(which(antisen))\n",
    "\n",
    "        sen        <-inrange(insites[iot[!is.na(iot[,1]),1],1],tmpann_2$start,tmpann_2$end)\n",
    "        antisen    <-inrange(insites[iot[!is.na(iot[,2]),2],2],tmpann_2$start,tmpann_2$end)\n",
    "        counts[l,3]<-length(which(sen)) + length(which(antisen))\n",
    "\n",
    "        sen        <-inrange(insites[iol[!is.na(iol[,1]),1],1],tmpann_2$start,tmpann_2$end)\n",
    "        antisen    <-inrange(insites[iol[!is.na(iol[,2]),2],2],tmpann_2$start,tmpann_2$end)\n",
    "        counts[l,4]<-length(which(sen)) + length(which(antisen))\n",
    "\n",
    "        gnames[l] <- jj\n",
    "\n",
    "        l <- l+1\n",
    "\n",
    "    }\n",
    "    print(proc.time() - ptm)\n",
    "    save(counts, gnames, file=paste0(outPath,ii,'.rda'))\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.5.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
