{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### In this notebook a data.table object is created holding human exon annotations corresponding to the gene annotation ./data/genes.rda"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colMeans,\n",
      "    colnames, colSums, dirname, do.call, duplicated, eval, evalq,\n",
      "    Filter, Find, get, grep, grepl, intersect, is.unsorted, lapply,\n",
      "    lengths, Map, mapply, match, mget, order, paste, pmax, pmax.int,\n",
      "    pmin, pmin.int, Position, rank, rbind, Reduce, rowMeans, rownames,\n",
      "    rowSums, sapply, setdiff, sort, table, tapply, union, unique,\n",
      "    unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following objects are masked from ‘package:data.table’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "The following object is masked from ‘package:data.table’:\n",
      "\n",
      "    shift\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "Loading required package: GenomicFeatures\n",
      "Loading required package: AnnotationDbi\n",
      "Loading required package: Biobase\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "Loading required package: AnnotationFilter\n",
      "\n",
      "Attaching package: 'ensembldb'\n",
      "\n",
      "The following object is masked from 'package:stats':\n",
      "\n",
      "    filter\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(data.table)\n",
    "library(ensembldb)\n",
    "load('./data/genes.rda') # Load gene annotation\n",
    "EnsDbFile <- './data/large_files/Homo_sapiens.GRCh38.94.sqlite' # Load an sqlite database object previously generated by ensembldb ensDbFromGFF() function\n",
    "edb <- EnsDb(EnsDbFile) # Create annotation database object from sqlite file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Get exon annotation\n",
    "exons <- exons(edb,columns=c('gene_name','gene_id','gene_biotype'))\n",
    "#--- Convert the annotation to data.table object\n",
    "exons <- data.table(chrom=as.vector(seqnames(exons)), start=start(exons), end=end(exons), gene_id=exons$gene_id, gene_sym=exons$gene_name, gene_biotype=exons$gene_biotype)\n",
    "#--- Make field for exon length\n",
    "exons$len <- exons$end-exons$start"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "702292"
      ],
      "text/latex": [
       "702292"
      ],
      "text/markdown": [
       "702292"
      ],
      "text/plain": [
       "[1] 702292"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#--- Filter for only exons in chromosomes 1-Y\n",
    "exons <- exons[exons$chrom %in% c(1:22,'X','Y'),]\n",
    "nrow(exons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "19896"
      ],
      "text/latex": [
       "19896"
      ],
      "text/markdown": [
       "19896"
      ],
      "text/plain": [
       "[1] 19896"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#--- Filter for only exons of genes included in gene annotation\n",
    "exons <- exons[exons$gene_id %in% genes$gene_id,]\n",
    "length(unique(exons$gene_id))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>chrom</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_id</th><th scope=col>gene_sym</th><th scope=col>gene_biotype</th><th scope=col>len</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>1              </td><td> 65419         </td><td> 65433         </td><td>ENSG00000186092</td><td>OR4F5          </td><td>protein_coding </td><td>  14           </td></tr>\n",
       "\t<tr><td>1              </td><td> 65520         </td><td> 65573         </td><td>ENSG00000186092</td><td>OR4F5          </td><td>protein_coding </td><td>  53           </td></tr>\n",
       "\t<tr><td>1              </td><td> 69037         </td><td> 71585         </td><td>ENSG00000186092</td><td>OR4F5          </td><td>protein_coding </td><td>2548           </td></tr>\n",
       "\t<tr><td>1              </td><td> 69055         </td><td> 70108         </td><td>ENSG00000186092</td><td>OR4F5          </td><td>protein_coding </td><td>1053           </td></tr>\n",
       "\t<tr><td>1              </td><td>450703         </td><td>451697         </td><td>ENSG00000284733</td><td>OR4F29         </td><td>protein_coding </td><td> 994           </td></tr>\n",
       "\t<tr><td>1              </td><td>685679         </td><td>686673         </td><td>ENSG00000284662</td><td>OR4F16         </td><td>protein_coding </td><td> 994           </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllllll}\n",
       " chrom & start & end & gene\\_id & gene\\_sym & gene\\_biotype & len\\\\\n",
       "\\hline\n",
       "\t 1                &  65419           &  65433           & ENSG00000186092  & OR4F5            & protein\\_coding &   14            \\\\\n",
       "\t 1                &  65520           &  65573           & ENSG00000186092  & OR4F5            & protein\\_coding &   53            \\\\\n",
       "\t 1                &  69037           &  71585           & ENSG00000186092  & OR4F5            & protein\\_coding & 2548            \\\\\n",
       "\t 1                &  69055           &  70108           & ENSG00000186092  & OR4F5            & protein\\_coding & 1053            \\\\\n",
       "\t 1                & 450703           & 451697           & ENSG00000284733  & OR4F29           & protein\\_coding &  994            \\\\\n",
       "\t 1                & 685679           & 686673           & ENSG00000284662  & OR4F16           & protein\\_coding &  994            \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "chrom | start | end | gene_id | gene_sym | gene_biotype | len | \n",
       "|---|---|---|---|---|---|\n",
       "| 1               |  65419          |  65433          | ENSG00000186092 | OR4F5           | protein_coding  |   14            | \n",
       "| 1               |  65520          |  65573          | ENSG00000186092 | OR4F5           | protein_coding  |   53            | \n",
       "| 1               |  69037          |  71585          | ENSG00000186092 | OR4F5           | protein_coding  | 2548            | \n",
       "| 1               |  69055          |  70108          | ENSG00000186092 | OR4F5           | protein_coding  | 1053            | \n",
       "| 1               | 450703          | 451697          | ENSG00000284733 | OR4F29          | protein_coding  |  994            | \n",
       "| 1               | 685679          | 686673          | ENSG00000284662 | OR4F16          | protein_coding  |  994            | \n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start  end    gene_id         gene_sym gene_biotype   len \n",
       "1 1      65419  65433 ENSG00000186092 OR4F5    protein_coding   14\n",
       "2 1      65520  65573 ENSG00000186092 OR4F5    protein_coding   53\n",
       "3 1      69037  71585 ENSG00000186092 OR4F5    protein_coding 2548\n",
       "4 1      69055  70108 ENSG00000186092 OR4F5    protein_coding 1053\n",
       "5 1     450703 451697 ENSG00000284733 OR4F29   protein_coding  994\n",
       "6 1     685679 686673 ENSG00000284662 OR4F16   protein_coding  994"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(exons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "save(exons,file='./data/exons.rda')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
