{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: BiocGenerics\n",
      "Loading required package: parallel\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colMeans,\n",
      "    colnames, colSums, dirname, do.call, duplicated, eval, evalq,\n",
      "    Filter, Find, get, grep, grepl, intersect, is.unsorted, lapply,\n",
      "    lengths, Map, mapply, match, mget, order, paste, pmax, pmax.int,\n",
      "    pmin, pmin.int, Position, rank, rbind, Reduce, rowMeans, rownames,\n",
      "    rowSums, sapply, setdiff, sort, table, tapply, union, unique,\n",
      "    unsplit, which, which.max, which.min\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "Loading required package: stats4\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "The following objects are masked from ‘package:data.table’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Attaching package: ‘IRanges’\n",
      "\n",
      "The following object is masked from ‘package:data.table’:\n",
      "\n",
      "    shift\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "Loading required package: GenomicFeatures\n",
      "Loading required package: AnnotationDbi\n",
      "Loading required package: Biobase\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "Loading required package: AnnotationFilter\n",
      "\n",
      "Attaching package: 'ensembldb'\n",
      "\n",
      "The following object is masked from 'package:stats':\n",
      "\n",
      "    filter\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(data.table)\n",
    "library(ensembldb)\n",
    "EnsDbFile <- './data/large_files/Homo_sapiens.GRCh38.94.sqlite' # Load an sqlite database object previously generated by ensembldb ensDbFromGFF() function\n",
    "edb <- EnsDb(EnsDbFile) # Create annotation database object from sqlite file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "58735"
      ],
      "text/latex": [
       "58735"
      ],
      "text/markdown": [
       "58735"
      ],
      "text/plain": [
       "[1] 58735"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#--- Get gene annotation\n",
    "genes <- genes(edb)\n",
    "#--- Convert the annotation to data.table object\n",
    "genes <- data.table(chrom=as.vector(seqnames(genes)), start=start(genes), end=end(genes), gene_id=genes$gene_id, gene_sym=genes$gene_name, gene_biotype=genes$gene_biotype)\n",
    "#--- Make field for gene length\n",
    "genes$len <- genes$end-genes$start\n",
    "#--- Print number of genes in annotation\n",
    "nrow(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "58639"
      ],
      "text/latex": [
       "58639"
      ],
      "text/markdown": [
       "58639"
      ],
      "text/plain": [
       "[1] 58639"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#--- Filter for only genes in chromosomes 1-Y\n",
    "genes <- genes[genes$chrom %in% c(1:22,'X','Y'),]\n",
    "nrow(genes)\n",
    "#--- Filter out any LRG genes\n",
    "# genes <- genes[grep('ENSG',genes$gene_id),]\n",
    "# nrow(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "57096"
      ],
      "text/latex": [
       "57096"
      ],
      "text/markdown": [
       "57096"
      ],
      "text/plain": [
       "[1] 57096"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#--- Print number of unique gene symbols\n",
    "length(unique(genes$gene_sym))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Filter annotation for only genes with biotype protein_coding\n",
    "genes <- genes[which(genes$gene_biotype=='protein_coding'),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "19909"
      ],
      "text/latex": [
       "19909"
      ],
      "text/markdown": [
       "19909"
      ],
      "text/plain": [
       "[1] 19909"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "19894"
      ],
      "text/latex": [
       "19894"
      ],
      "text/markdown": [
       "19894"
      ],
      "text/plain": [
       "[1] 19894"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#--- Print number of genes\n",
    "length(genes$gene_sym)\n",
    "#--- Print number of unique gene symbols\n",
    "length(unique(genes$gene_sym))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# --- Print annotation of genes with duplicate symbols\n",
    "# genes[genes$gene_sym %in% genes$gene_sym[duplicated(genes$gene_sym)],]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Create a list of Ensembl IDs to remove corresponding to duplicated gene symbols with overlapping ranges amongst duplicates\n",
    "todel <- c()\n",
    "for (ii in genes$gene_sym[duplicated(genes$gene_sym)]) {\n",
    "    tmp <- genes[genes$gene_sym == ii,]\n",
    "    tmp <- tmp[order(-tmp$len),]\n",
    "    if (length(intersect(tmp$start[1]:tmp$end[1],tmp$start[2]:tmp$end[2]))>0) {\n",
    "        todel <- append(todel,tail(tmp$gene_id,nrow(tmp)-1))\n",
    "    }\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "#--- Filter out overlapping duplicate genes\n",
    "genes <- genes[!(genes$gene_id %in% todel),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "19896"
      ],
      "text/latex": [
       "19896"
      ],
      "text/markdown": [
       "19896"
      ],
      "text/plain": [
       "[1] 19896"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nrow(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table>\n",
       "<thead><tr><th scope=col>chrom</th><th scope=col>start</th><th scope=col>end</th><th scope=col>gene_id</th><th scope=col>gene_sym</th><th scope=col>gene_biotype</th><th scope=col>len</th></tr></thead>\n",
       "<tbody>\n",
       "\t<tr><td>1              </td><td> 65419         </td><td> 71585         </td><td>ENSG00000186092</td><td>OR4F5          </td><td>protein_coding </td><td> 6166          </td></tr>\n",
       "\t<tr><td>1              </td><td>450703         </td><td>451697         </td><td>ENSG00000284733</td><td>OR4F29         </td><td>protein_coding </td><td>  994          </td></tr>\n",
       "\t<tr><td>1              </td><td>685679         </td><td>686673         </td><td>ENSG00000284662</td><td>OR4F16         </td><td>protein_coding </td><td>  994          </td></tr>\n",
       "\t<tr><td>1              </td><td>923928         </td><td>944581         </td><td>ENSG00000187634</td><td>SAMD11         </td><td>protein_coding </td><td>20653          </td></tr>\n",
       "\t<tr><td>1              </td><td>944204         </td><td>959309         </td><td>ENSG00000188976</td><td>NOC2L          </td><td>protein_coding </td><td>15105          </td></tr>\n",
       "\t<tr><td>1              </td><td>960587         </td><td>965715         </td><td>ENSG00000187961</td><td>KLHL17         </td><td>protein_coding </td><td> 5128          </td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "\\begin{tabular}{r|lllllll}\n",
       " chrom & start & end & gene\\_id & gene\\_sym & gene\\_biotype & len\\\\\n",
       "\\hline\n",
       "\t 1                &  65419           &  71585           & ENSG00000186092  & OR4F5            & protein\\_coding &  6166           \\\\\n",
       "\t 1                & 450703           & 451697           & ENSG00000284733  & OR4F29           & protein\\_coding &   994           \\\\\n",
       "\t 1                & 685679           & 686673           & ENSG00000284662  & OR4F16           & protein\\_coding &   994           \\\\\n",
       "\t 1                & 923928           & 944581           & ENSG00000187634  & SAMD11           & protein\\_coding & 20653           \\\\\n",
       "\t 1                & 944204           & 959309           & ENSG00000188976  & NOC2L            & protein\\_coding & 15105           \\\\\n",
       "\t 1                & 960587           & 965715           & ENSG00000187961  & KLHL17           & protein\\_coding &  5128           \\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "chrom | start | end | gene_id | gene_sym | gene_biotype | len | \n",
       "|---|---|---|---|---|---|\n",
       "| 1               |  65419          |  71585          | ENSG00000186092 | OR4F5           | protein_coding  |  6166           | \n",
       "| 1               | 450703          | 451697          | ENSG00000284733 | OR4F29          | protein_coding  |   994           | \n",
       "| 1               | 685679          | 686673          | ENSG00000284662 | OR4F16          | protein_coding  |   994           | \n",
       "| 1               | 923928          | 944581          | ENSG00000187634 | SAMD11          | protein_coding  | 20653           | \n",
       "| 1               | 944204          | 959309          | ENSG00000188976 | NOC2L           | protein_coding  | 15105           | \n",
       "| 1               | 960587          | 965715          | ENSG00000187961 | KLHL17          | protein_coding  |  5128           | \n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  chrom start  end    gene_id         gene_sym gene_biotype   len  \n",
       "1 1      65419  71585 ENSG00000186092 OR4F5    protein_coding  6166\n",
       "2 1     450703 451697 ENSG00000284733 OR4F29   protein_coding   994\n",
       "3 1     685679 686673 ENSG00000284662 OR4F16   protein_coding   994\n",
       "4 1     923928 944581 ENSG00000187634 SAMD11   protein_coding 20653\n",
       "5 1     944204 959309 ENSG00000188976 NOC2L    protein_coding 15105\n",
       "6 1     960587 965715 ENSG00000187961 KLHL17   protein_coding  5128"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(genes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "save(genes,file='./data/genes.rda')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
