# analyses

#### sim_out_analyses
* Contains analyses of simulation results.

#### tsg_insertion_statistics
* Statistical analyses of sites within TSGs.

#### gene_lists
* Lists and statistics of frequently mutated genes in cases of lung, colon and brain cancer from TCGA.

#### cancer_type_site_counts.ipynb
* Notebook to estimate the probability of driver, passenger, and null (no effect) L1 insertions across lung, colon, and brain cancer.

#### count_cds_sites.ipynb
* Notebook that uses the hg38 gff3 file to compute the number of L1 target sites of each SV category in known coding regions and non-coding regions of the genome (separately), and also the coding fraction of each chromosome.

#### count_exonic_sites.ipynb
* Equivalent to cds_insertions.ipynb, except for exonic regions rather than just coding regions.

#### count_gene_sites_all.ipynb
* Uses the TxDb.Hsapiens.UCSC.hg38.knownGene, BSgenome.Hsapiens.UCSC.hg38, and org.Hs.eg.db bioconductor libraries to generate a table containing, for each known gene, the number of L1 target sites of each SV category, as well as the gene length.

#### count_gene_sites_exonic.r
* Uses the table exann.rda (output by sim-develop/src/sim_exon_annotation.ipynb) to determine the number of sites in the exonic regions of each gene in hg38.

#### count_sites_exonicvsnon.ipynb
* Notebook to count the distribution of L1 target loci in exonic vs. non-exonic regions of the human genome (hg38).

#### gene_site_distribution.ipynb
* The table generated by "count_sites_in_genes.ipynb" is used to plot the distribution of L1 target sites in known genes.

#### stats_of_codingRegion_insertions.xlsx
* Excel file containing further statistics from data generated by count_cds_sites.ipynb and count_exonic_sites.ipynb
