# analyses
Statistical analyses of L1 EN sites in hg38 for development of L1 insertion models.

#### SimResAn
* Contains analyses of simulation results.

#### tsg_insertion_statistics
* Statistical analyses of sites within TSGs.

#### compute_mutation_type_probabilities.ipynb
* The table exann.rda (output by sim-develop/src/sim_exon_annotation.ipynb), and the L1 target site maps of each chromosome (output by ~/jackgl/sim-develop/src/mapgenome.r) are used to compute the number of sites of each SV category in TSG exons, non-TSG exons, and non-exonic regions of hg38, and then the probabilities of passenger and driver type mutations.

#### count_cds_sites.ipynb
* Notebook that uses the hg38 gff3 file to compute the number of L1 target sites of each SV category in known coding regions and non-coding regions of the genome (separately), and also the coding fraction of each chromosome.

#### count_exonic_sites.ipynb
* Equivalent to cds_insertions.ipynb, except for exonic regions rather than just coding regions.

#### count_gene_sites_all.ipynb
* Uses the TxDb.Hsapiens.UCSC.hg38.knownGene, BSgenome.Hsapiens.UCSC.hg38, and org.Hs.eg.db bioconductor libraries to generate a table containing, for each known gene, the number of L1 target sites of each SV category, as well as the gene length.

#### gene_site_distribution.ipynb
* The table generated by "count_sites_in_genes.ipynb" is used to plot the distribution of L1 target sites in known genes.

#### count_gene_sites_exonic.r
* Uses the table exann.rda (output by sim-develop/src/sim_exon_annotation.ipynb) to determine the number of sites in the exonic regions of each gene in hg38.

#### exon_insertionstats.r
* Does the same thing as exon_insertions.ipynb, except uses exann.rda (output by sim-develop/src/sim_exon_annotation.ipynb) rather than the hg38 gff3 file.

#### exon_simstats.r
* Counts the number of simulated insertions in each SV category that lie within exons. It is meant to be used with the v1 simulation.

#### gene_exon_site_counts.rda
* Output by count_gene_sites_exonic.r

#### gene_site_counts.rda
* Output by count_gene_sites_all.ipynb

#### stats_of_codingRegion_insertions.xlsx
* Excel file containing further statistics from data generated by count_cds_sites.ipynb and count_exonic_sites.ipynb
